modules/frontend/tracebyidsharding.go

```diff
@@ -65,7 +65,10 @@ func (s shardQuery) RoundTrip(r *http.Request) (*http.Response, error) {
 
 	// context propagation
 	r = r.WithContext(ctx)
-	reqs, err := s.buildShardedRequests(r)
+	subCtx, subCancel := context.WithCancel(ctx)
+	defer subCancel()
+
```

Comment:
nit: if it's not too hard to test, can we have test to test the cancellation behavour. test would ensure that we don't break it again :)

Reply:
I added a test case, but its only against eh mock.  This is challenging to test e2e.  How does the test look to you?  Do you see a simpler path to testing this change?

Reply:
I backed out the test case since it was requested we don't change the status handling.  I don't think there is good way to test this.